pkg/controllers/node/termination/suite_test.go

```diff
@@ -99,25 +101,37 @@ var _ = Describe("Termination", func() {
 
 	Context("Reconciliation", func() {
 		It("should delete nodes", func() {
-			ExpectApplied(ctx, env.Client, node)
+			ExpectApplied(ctx, env.Client, node, nodeClaim)
 			Expect(env.Client.Delete(ctx, node)).To(Succeed())
 			node = ExpectNodeExists(ctx, env.Client, node.Name)
-			ExpectObjectReconciled(ctx, env.Client, terminationController, node)
-			ExpectNotFound(ctx, env.Client, node)
+			Eventually(func(g Gomega) {
```

Comment:
I'd like us to avoid doing eventual calls when we control what the backend looks like. Can we just make the number of calls that we need to make to ensure that the instance is removed on the backend so that we know a specific ExpectObjectReconciled call will succeed to terminate?

Reply:
These kinds of eventually calls also lead to the kind of state that we are seeing over in the disruption code where we don't have any kind of guarantees that operations will be completed if we are relying on a time construct

Reply:
We could, currently we only need two iterations. I had opted for eventually since I didn't see a massive increase in test duration and this decoupled the test from the fake cloudprovider implementation. I'm fine with hardcoding this to two iterations and adjusting if we need to later down the road.

Reply:
I think we should opt for determinism any time that we can. Let's go with two reconciliations for now